src/Dependencies/PooledObjects/PooledDelegates.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.PooledObjects
+{
+    internal static class PooledDelegates
```

Comment:
>PooledDelegates [](start = 26, length = 15)

It feels like this class and its members can benefit from comments about when and how they are supposed to be used.

Reply:
Added a bunch of comments.